Source/WebCore/platform/text/cocoa/LocaleCocoa.mm

```diff
@@ -245,26 +245,43 @@
 
 #endif
 
-using CanonicalLocaleMap = HashMap<AtomString, AtomString>;
+using CanonicalLocaleMap = HashMap<AtomString, RetainPtr<CFStringRef>>;
 
-static CanonicalLocaleMap& canonicalLocaleMap()
+struct LocaleCache {
+    AtomString m_key;
+    RetainPtr<CFStringRef> m_value;
+    CanonicalLocaleMap m_map;
+};
+
+
+static LocaleCache& localeCache()
 {
-    static NeverDestroyed<CanonicalLocaleMap> canonicalLocaleMap;
-    return canonicalLocaleMap.get();
+    static NeverDestroyed<LocaleCache> localeCache;
```

Comment:
nit: Would be nice to switch to `MainThreadNeverDestroyed` since you're updating this line.

Reply:
Changed.